tests/foreman/ui/test_rhcloud_insights.py

```diff
@@ -496,3 +497,67 @@ def test_insights_tab_on_host_details_page(
                 assert recommendation['label'] == 'Moderate'
                 assert dnf_issue in recommendation['text']
                 assert len(insights_recommendations) == int(result['Recommendations'])
+
+
+@pytest.mark.run_in_one_thread
+@pytest.mark.no_containers
+def test_insights_registration_with_capsule(
+    rhcloud_capsule, organization_ak_setup, rhcloud_sat_host, rhel7_contenthost, default_os
+):
+    """Registering host with insights having traffic going through
+        external capsule.
+
+    :id: 9db1d307-664c-4d4a-89de-da986224f071
+
+    :customerscenario: true
+
+    :steps:
+        1. Integrate a capsule with satellite.
+        2. open the global registration form and select the same capsule.
+        3. Overide Insights and Rex parameters.
+        4. check host is registered successfully with selected capsule.
+        5. Test insights client connection & reporting status.
+
+    :expectedresults: Host is successfully registered with capsule host,
+             having remote execution and insights.
+
+
+    :CaseImportance: High
+
+    :BZ: 2110222
+
+    :CaseAutomation: Automated
+    """
+    org, ak = organization_ak_setup
+    capsule = rhcloud_sat_host.api.SmartProxy(name=rhcloud_capsule.hostname).search()[0]
+    org.smart_proxy.append(capsule)
+    org.update(['smart_proxy'])
+
+    with Session(hostname=rhcloud_sat_host.hostname) as session:
+        session.organization.select(org_name=org.name)
+        session.location.select(loc_name=DEFAULT_LOC)
+        cmd = session.host.get_register_command(
+            {
+                'general.operating_system': default_os.title,
+                'general.orgnization': org.name,
+                'general.capsule': rhcloud_capsule.hostname,
```

Comment:
Can we checkout 2 chosts here, and cover the scenario for registering chosts on satellite and capsule both to install insights-client?

Reply:
The main plan is to have all the rhel content hosts passed through the external capsule, which is not possible immediately as we are dealing with the BZ opened for that. I will be parameterizing these tests. And also we have many tests going through sat for insights-client.